docs/docs/spaces.mdx

```diff
@@ -0,0 +1,237 @@
+
+### What happens during the assembly?
+
+The most important step during the assembly is the prefixing. During this step every
+intent, entity, slot, action, form, utterance that is defined in a space's domain file
+is prefixed (infixed for utterances) with this space's name. For example, an intent
+`ask_transfer_charge` in the `transfer_money` domain would become
+`transfer_money.ask_transfer_charge` and every reference of this intent would be
+adjusted. The final assistant then works on the prefixed data.
+
+An exception to this is the main space. Anything in the main space and all its
+primitives that are used in other spaces, will not be prefixed.
+
+Another exception are rules. They don't have a name that can be prefixed. Instead,
+we add a condition to each rule that it is only applicable while it's space is active.
+
+### How is space activation tracked?
+
+Space activation is currently tracked through boolean slots that are automatically
+generated during assembly for each space.
+
+### How does filter and rerank work?
+
+The filter and rerank component post-processes the intent ranking of the intent
+classification components. It accesses the tracker and checks which space are active
+or would be activated, in case an entry intent is at the top.
+
+It then removes any intents from the ranking that are not possible. Further it also
+removes any entities that are not possible given the space activation status or
+about to be predicted entry intents.
+
+### How does entity recognition work differently?
+
+Usually, entity recognizers in Rasa only return a single label per token in a message.
+Thus, there is no ranking, that could be post-processed as in the case of intents. We
+have built our `SpacesCRFEntityExtractor` in a way that it creates multiple extractors.
+One for each space. Now, during the post-processing step, we can filter out the
+extractor of the spaces that are not activated.
+
+### Custom Actions
+
+Custom actions work as before. However, the tracker, the domain, and slots
+will be stripped of any information from other spaces before being handed to your
+action. Additionally, every event such as slot sets, will be prefixed after they are
+returned by your action. All of this ensures that from the view point of your custom
+action, you don't need to worry about the other spaces and accidentally leaking
+or altering information. It is as if the other spaces and prefixing does not exist.
+
+### Response Selection
+
+Response selection works as before. The only small difference is that in your
+`config.yaml` you'll need to specify the retrieval intent including it's final prefix.
+If your retrieval intent is `investment_faq` in the `investment` space, then in your
+config you'll need to set `investment.investment_faq` as the retrieval intent. If the
+retrieval intent belongs to the main space, no prefix is added. That's why
+, in this case, during the definition of the retrieval intent, no prefixing is necessary.
+
```

Comment:
can we also mention synonyms and lookup tables, together with limitations if any?

Reply:
there were non for the synonyms, right? Are there any for the lookup tables @Shmuma ?

Reply:
but at least mentioning that the feature works with both of them (no limitation for lookup tables; some limitations for synonyms) would be good. I've already described those in [this section](https://www.notion.so/rasa/Spaces-Implementation-Proposal-updated-Q2-2023-1a834e7203cb47cdabfb46049a0c983d?pvs=4#12c7d0c51a3d49f6ad9eb49c40e4f8eb)